packages/api-server/src/plugins/trajectory/index.ts

```diff
@@ -0,0 +1,145 @@
+import * as assert from 'assert';
+import WebSocket from 'ws';
+import { Argv } from 'yargs';
+import ApiGateway from '../../api-gateway';
+import logger from '../../logger';
+
+export function options(yargs: Argv) {
+  return yargs.option('trajectoryServerUrl', {
+    type: 'string',
+    demandOption: true,
+  });
+}
+
+type ConfigType = ReturnType<typeof options>['argv'];
+
+export async function onLoad(config: ConfigType, api: ApiGateway): Promise<void> {
+  const plugin = new TrajectoryPlugin(config);
+  api.registerHandler('latestTrajectory', (params) => plugin.latestTrajectory(params));
+  api.registerHandler('trajectoryServerTime', (params) => plugin.serverTime(params));
+}
+
+interface TrajectoryRequest {
+  request: 'trajectory';
+  mapName: string;
+  duration: number;
+  trim: boolean;
+}
+
+export type LatestTrajectoryParams = Omit<TrajectoryRequest, 'request'>;
+
+// RawVelocity received from server is in this format (x, y, theta)
+export type RawVelocity = [number, number, number];
+
+// RawPose2D received from server is in this format (x, y, theta)
+export type RawPose2D = [number, number, number];
+
+export interface RawKnot {
+  t: number; // milliseconds
+  v: RawVelocity;
+  x: RawPose2D;
+}
+
+export interface Trajectory {
+  id: number;
+  shape: string;
+  dimensions: number;
+  segments: RawKnot[];
+  robot_name: string;
+  fleet_name: string;
+}
+
+export type Conflict = number[];
```

Comment:
Not for this PR but, should we use just these definitions in the `dashboard/robot-trajectory-manager.ts` too? to keep one source of truth

Reply:
yea, these are copied from that file. When we change the trajectory client to use the api server we should have it reference the typings here.